### PR TITLE
Adding new actions (4) and new rule 

### PR DESCRIPTION
**Pull request by nullbuddy1243**
New actions: 
- **device42.get_device_by_id:** gets a device by id

- **device42.get_lifecycle_events:** fetches lifecycle events with optional filtering by date, asset number, device names, lifecycle type, and enduser name.

- **device42.udpate_object_category_by_lifecycle_id:** Takes input of a lifecycle event type ID and applies a new object category, or other arbitrary changes, to a device as specified by the object category stored in the st2 datastore under {lifecycle_type_id}_lc.  Utilized in `lifecycle_triggered_object_category_change` rule. 

- **device42.update_device:**  Arbitrarily updates a device on D42. 

New Rule: 
- **device42.lifecycle_triggered_object_category_change**: implements an API endpoint for D42 to point lifecycle event webhooks at. Triggers the `udpate_object_category_by_lifecycle_id` action whenever a lifecycle event occurs in D42. 

contact author @ will.acheson@device42.com

### CODE REVIEW

**Comment by LindsayHill** [comment]
Thanks for the PR @deusofnull, much appreciated. Looks like a few standard linting things, nothing too major. Could you tidy those up?

One tip: it's usually faster to run flake8 & pylint locally, using our configs from github.com/StackStorm/lint-configs. That lets you find & fix all the lint issues faster, since if you update this PR, CircleCI will fail on the first error.

I found out that this gets tedious if you have issues in multiple files :)

**Comment by nullbuddy1243** [comment]
@LindsayHill  thanks for the response!  I am working locally to get all these formatting issues solved.  Hopefully my recent commits fix them.  Im using VS code's flake8 linter right now, so if that doesnt work, Ill install the stackstorm linter proper.

**Comment by nullbuddy1243** [comment]
Success!  This online debugger console is very nifty, by the way.

File: README.md [hunk 1]
```diff
@@ -22,13 +22,23 @@ You can also use dynamic values from the datastore. See the
...
+| device42.device_name_list                | device42 | Returns list of devices names            |
+| device42.get_device_by_id                | device42 | Get a device with full details from D42  |
+|                                          |          | by its ID                                |
+| device42.get_dns_zone                    | device42 | Returns DNS zone file                    |
+| device42.get_lifecycle_events            | device42 | Get lifecycle events from D42 with       |
+|                                          |          | optional filtering parameters.           |
+|                                          |          | https://api.device42.com/#asset-device-  |
+|                                          |          | life-cycle                               |
+| device42.suggest_next_ip                 | device42 | Suggest next available IP Address        |
+| device42.udpate_object_category_by_lifec | device42 | Update a devices object category and     |
```
Comment (LindsayHill) [inline README.md]: Looks like typo here? `udpate_object...`

File: actions/lc_objcat_updater.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+---
+name: udpate_object_category_by_lifecycle_id
```
Comment (LindsayHill) [inline actions/lc_objcat_updater.yaml]: Ah, here's the source of the typo

File: rules/lifecycle_triggered_object_category_change.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    description: "Respond to lifecycle event webhooks whenever a device gets a new lifecycle.  When this happens, update that devices object category. "
+    enabled: true
+
+    trigger:
+        type: "core.st2.webhook"
+        parameters:
+            url: "d42_lifecycle"
+
+    action:
+        ref: "device42fork.udpate_object_category_by_lifecycle_id"
```
Comment (LindsayHill) [inline rules/lifecycle_triggered_object_category_change.yaml]: This one will also need fixing, and a few lines below here

**Review by LindsayHill — APPROVED** [review]
PR generally looks good, apart from minor typo that occurs in a few places. 

Could you fix that, and add an entry to CHANGES.md?

Then I can merge it

**Comment by nullbuddy1243** [comment]
@LindsayHill  Sure thing.

**Comment by nullbuddy1243** [comment]
@LindsayHill that should take care of it.  Let me know if there's anything else.

**Comment by LindsayHill** [comment]
@deusofnull looks like you've still got a few `s/udpate/update/` corrections to make?

**Comment by nullbuddy1243** [comment]
@LindsayHill I am a bit confused.  What is the typo in README.md?  Do you mean that there's an issue from the line for `device42.update_object_category_by_lifecycle_id`?  i.e because it overflows to a second line?

**Comment by LindsayHill** [comment]
Check the `name` field in lc_objcat_updater.yaml - you have `udpate` in there, should be `update`

**Comment by nullbuddy1243** [comment]
Ok, wow my brain just did not recognize that until just now.  D'oh!  Submitting fix in a second.

**Review by LindsayHill — APPROVED** [review]

**Comment by LindsayHill** [comment]
Thanks @deusofnull. Sometimes you get to close to something and your eyes skip over it :)

**Comment by nullbuddy1243** [comment]
thanks @LindsayHill !

**Pull request merged**